Address new deprecations in SciPy 1.12.0.dev0

Our nightly test job is doing its job and flagging pending deprecation warnings: https://github.com/scikit-image/scikit-image/actions/runs/5485938623/jobs/9995372984#step:5:3983

```
FAILED segmentation/tests/test_random_walker.py::test_2d_cg[float16] - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_2d_cg[float32] - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_2d_cg[float64] - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_2d_cg_mg[float16] - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_2d_cg_mg[float32] - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_2d_cg_mg[float64] - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_types - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_3d[float32] - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_3d[float64] - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_3d_inactive - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_multispectral_2d[float32-0] - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_multispectral_2d[float32-1] - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_multispectral_2d[float32--1] - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_multispectral_2d[float64-0] - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_multispectral_2d[float64-1] - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_multispectral_2d[float64--1] - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_multispectral_3d[float32] - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_multispectral_3d[float64] - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_spacing_0 - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_spacing_1 - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_isolated_seeds - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_isolated_area - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
FAILED segmentation/tests/test_random_walker.py::test_prob_tol - ValueError: Unexpected warning: 'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in favor of 'rtol' and will be removed in SciPy v.1.14.0. Until then, if set, it will override 'rtol'.
```

Don't have time right to look into it right now.

--- a/skimage/_shared/testing.py
+++ b/skimage/_shared/testing.py
@@ -280,6 +280,17 @@ def setup_test():
             category=UserWarning,
         )
 
+        # Temporary warning raised by scipy. May be removed when scipy 1.12 is
+        # minimum supported version and we replace tol with rtol
+        warnings.filterwarnings(
+            "default",
+            message=(
+                "'scipy.sparse.linalg.cg' keyword argument 'tol' is deprecated in "
+                "favor of 'rtol' and will be removed in SciPy v.1.14.0."
+            ),
+            category=DeprecationWarning,
+        )
+
 
         warnings.filterwarnings(
             "default",

